lib/internal/streams/add-abort-signal.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+const {
+  AbortError,
+  codes,
+} = require('internal/errors');
+
+const { ERR_INVALID_ARG_TYPE } = codes;
+
+const {
+  validateAbortSignal,
+} = require('internal/validators');
+
+function isReadable(obj) {
+  return !!(obj && typeof obj.pipe === 'function');
+}
```

Comment:
I don't think this works as you think it would. I think this passes for every descendant of Stream.

Reply:
I think so too and so do the tests - but I just copied `isReadable` from `pipeline`